inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java

```diff
@@ -0,0 +1,660 @@
+        }
+    }
+
+    /**
+     * If this function is called successively without calling {@see #destroyConnection()}, only the
+     * first call has any effect.
+     *
+     * @throws FlumeException if an RPC client connection could not be opened
+     */
+    private void createConnection() throws FlumeException {
+        // if already connected, just skip
+        if (sessionFactory != null) {
+            return;
+        }
+
+        try {
+            TubeClientConfig conf = initTubeConfig(masterHostAndPortList);
+            sessionFactory = new TubeMultiSessionFactory(conf);
+        } catch (TubeClientException e) {
+            logger.error("create connnection error in tubesink, "
+                    + "maybe tube master set error, please re-check. ex1 {}", e.getMessage());
+            throw new FlumeException("connect to Tube error1, "
+                    + "maybe zkstr/zkroot set error, please re-check");
+        } catch (Throwable e) {
+            logger.error("create connnection error in tubesink, "
+                            + "maybe tube master set error/shutdown in progress, please re-check. ex2 {}",
+                    e.getMessage());
+            throw new FlumeException("connect to meta error2, "
+                    + "maybe tube master set error/shutdown in progress, please re-check");
+        }
+
+        if (producerMap == null) {
+            producerMap = new HashMap<String, MessageProducer>();
+        }
+        logger.debug("create tube connection");
+    }
+
+    private TubeClientConfig initTubeConfig(String masterHostAndPortList) throws Exception {
+        final TubeClientConfig tubeClientConfig = new TubeClientConfig(masterHostAndPortList);
+        tubeClientConfig.setLinkMaxAllowedDelayedMsgCount(linkMaxAllowedDelayedMsgCount);
+        tubeClientConfig.setSessionWarnDelayedMsgCount(sessionWarnDelayedMsgCount);
+        tubeClientConfig.setSessionMaxAllowedDelayedMsgCount(sessionMaxAllowedDelayedMsgCount);
+        tubeClientConfig.setNettyWriteBufferHighWaterMark(nettyWriteBufferHighWaterMark);
+        tubeClientConfig.setHeartbeatPeriodMs(15000L);
+        tubeClientConfig.setRpcTimeoutMs(20000L);
+
+        return tubeClientConfig;
+    }
+
+    private void destroyConnection() {
+        for (Map.Entry<String, MessageProducer> entry : producerMap.entrySet()) {
+            MessageProducer producer = entry.getValue();
+            try {
+                producer.shutdown();
+            } catch (TubeClientException e) {
+                logger.error("destroy producer error in tubesink, MetaClientException {}", e.getMessage());
+            } catch (Throwable e) {
+                logger.error("destroy producer error in tubesink, ex {}", e.getMessage());
+            }
+        }
+        producerMap.clear();
+
+        if (sessionFactory != null) {
+            try {
+                sessionFactory.shutdown();
+            } catch (TubeClientException e) {
```

Comment:
If different exceptions are handled the same way, using one `catch (Exception e)` could be better.